File: Online Judges/LeetCode/June LeetCoding Challenge/Week-2/Sort Colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        low, high = 0, len(nums)-1

        for pos in range(0, high+1):
            while nums[pos] == 2 and pos < high:
                nums[high], nums[pos] = nums[pos], nums[high]
                high -= 1

            while nums[pos] == 0 and pos >= low:
                nums[low], nums[pos] = nums[pos], nums[low]
                low += 1

File: Online Judges/LeetCode/June LeetCoding Challenge/Week-2/Sort Colors.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        if nums.len() <= 1 {
            return;
        }
        
        let mut i = 0;
        let mut j = 0;
        let mut k = nums.len() - 1;
        
        while j <= k && k > 0 {
            if nums[j] == 0 {
                nums.swap(i, j);
                i += 1;
                j += 1;
            } else if nums[j] == 1 {
                j += 1;
            } else {
                nums.swap(j, k);
                k -= 1;
            }
        }
    }
}